Amazon.QLDB.Driver/session/QldbSession.cs

```diff
@@ -149,7 +149,8 @@ public T Execute<T>(Func<TransactionExecutor, T> func, Action<int> retryAction)
                 }
                 catch (InvalidSessionException ise)
                 {
-                    this.isClosed = true;
+                    this.isDisposed = true;
+                    this.Destroy();
                     throw ise;
                 }
                 catch (TransactionAbortedException ae)
```

Comment:
can we rename the variable to tae

Reply:
It seems like theres a bug in github, I can't seem to add comments in the code following this comment..

Anyways,

Can we set `this.isDisposed = true` after L174

Reply:
In QldbSession.Destroy() we need to add `this.disposeDelegate(null); ` so it is returned back to pool

Reply:
We need to apply the same logic to TransactionAlreadyOpenException
```
this.isDisposed = true;
this.Destroy();
```

on L173

Reply:
TransactionAlreadyOpenException does not need to destroy the session. It can abort the transaction and start again with a new getSession() call..